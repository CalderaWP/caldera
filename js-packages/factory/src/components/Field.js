import React, {Fragment} from 'react';
import {fieldFactory} from '../factories/fieldFactory'

export const Field = ({
							   field, onChange,onBlur
						   }) => (
	<Fragment>{fieldFactory(field, onChange)}</Fragment>
);
