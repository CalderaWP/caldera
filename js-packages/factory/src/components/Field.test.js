import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField
} from '../fields.fixtures';

import {Field} from './Field';

describe( 'Field component', () => {
	let onChange;
	let onBlur;
	beforeEach( () => {
		onChange= jest.fn();
		onBlur= jest.fn()
	});

	it( 'Creates a text field', () => {
		const component = renderer.create(
			<Field
				field={textField} onChange={onChange} onBlur={onBlur}
		/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
	it( 'Creates a number field', () => {
		const component = renderer.create(
			<Field
				field={numberField} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Creates an email field', () => {
		const component = renderer.create(
			<Field
				field={emailField} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Creates an checkbox field field', () => {
		const component = mount(
			<Field
				field={checkboxField} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.find('input').prop( 'type')).toEqual('checkbox');
	});

	it( 'Creates an checkbox field set', () => {
		const component = renderer.create(
			<Field
				field={checkboxFieldset} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Creates an select field', () => {
		const component = renderer.create(
			<Field
				field={selectField} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Creates an select field identified as dropdown', () => {
		const component = renderer.create(
			<Field
				field={{
					...selectField,
					fieldType:'dropdown'
				}}
				onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
