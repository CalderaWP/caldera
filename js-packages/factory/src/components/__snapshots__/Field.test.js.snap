// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component Creates a text field 1`] = `
<div
  className="caldera-field caldera-field-text"
>
  <label
    className="caldera-field-label caldera-field-label-undefined"
    htmlFor="firstName"
  >
    First Name
  </label>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby={undefined}
        className="components-text-control__input"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value="roy"
      />
    </div>
  </div>
</div>
`;
